### ISSUE TITLE
Training new data without training old data and not forgetting old data.

### ISSUE THREAD

**Issue by DoanVanQuan21** [ultralytics/ultralytics#14985]
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

I'm using YOLOv8. 
Suppose, I have a trained animal_detect dataset of 500 dog images, and now I want to add 500 images including 100 dog images and 400 cat images in animal_detect and I just want to use the trained results Initial 500 images to continue training the next 500 images without having to retrain 1000 images.

Can someone give me a solution to solve this problem?

**Comment by pderrenger** [comment]
@DoanVanQuan21 to continue training your YOLOv8 model with new data without forgetting the old data, you can use a technique called "incremental learning" or "fine-tuning." Start by loading your pre-trained model and then train it further on the new dataset. This approach allows the model to retain knowledge from the initial 500 images while learning from the additional 500 images. Ensure that your new dataset includes both the old and new classes to maintain the model's performance across all classes. If you encounter any issues, please verify that you are using the latest version of the Ultralytics package.

**Comment by Y-T-G** [comment]
Not possible.

**Comment by DoanVanQuan21** [comment]
> @DoanVanQuan21 to continue training your YOLOv8 model with new data without forgetting the old data, you can use a technique called "incremental learning" or "fine-tuning." Start by loading your pre-trained model and then train it further on the new dataset. This approach allows the model to retain knowledge from the initial 500 images while learning from the additional 500 images. Ensure that your new dataset includes both the old and new classes to maintain the model's performance across all classes. If you encounter any issues, please verify that you are using the latest version of the Ultralytics package.

Hi, I tried using "incremental learning" but it seems that after I trained 500 more images including 100 dogs and 400 cats, the dog detection confidence decreased compared to the previous training.
This is my code

    # Load the pretrained model
    model = YOLO('./runs/detect/animal_detect/weights/best.pt')

    # Train the model
    model.train(data=os.path.join(dataset_path, 'data.yaml'), epochs=150, imgsz=640, batch=8, device=[0],lr0=0.00001)
   
Is there a problem with my code?
Can you help me check it?
Thanks.

**Comment by pderrenger** [comment]
It sounds like you're experiencing a common issue with incremental learning where the model's performance on previously learned classes degrades. This is known as "catastrophic forgetting." To mitigate this, you can try a few strategies:

1. Ensure your new dataset is balanced and includes a sufficient number of images for all classes, including the old ones.
2. Use a lower learning rate to fine-tune the model gently.
3. Consider using techniques like knowledge distillation or regularization to help retain the old knowledge.

Your code looks generally correct, but you might want to experiment with different learning rates or training parameters. Also, make sure you're using the latest version of the Ultralytics package to benefit from any recent improvements or bug fixes. If the issue persists, please provide more details about your dataset and training process.

**Comment by DoanVanQuan21** [comment]
> It sounds like you're experiencing a common issue with incremental learning where the model's performance on previously learned classes degrades. This is known as "catastrophic forgetting." To mitigate this, you can try a few strategies:
> 
> 1. Ensure your new dataset is balanced and includes a sufficient number of images for all classes, including the old ones.
> 2. Use a lower learning rate to fine-tune the model gently.
> 3. Consider using techniques like knowledge distillation or regularization to help retain the old knowledge.
> 
> Your code looks generally correct, but you might want to experiment with different learning rates or training parameters. Also, make sure you're using the latest version of the Ultralytics package to benefit from any recent improvements or bug fixes. If the issue persists, please provide more details about your dataset and training process.

Thank you for helping me.
Best regards.

**Comment by pderrenger** [comment]
You're welcome! To address the issue of decreased dog detection confidence, ensure your new dataset is well-balanced and includes sufficient images for all classes. Additionally, using a lower learning rate for fine-tuning can help. If you haven't already, please update to the latest version of the Ultralytics package to benefit from recent improvements. If the problem persists, providing more details about your dataset and training process could help diagnose the issue further.

**Comment by omeganoob** [comment]
> You're welcome! To address the issue of decreased dog detection confidence, ensure your new dataset is well-balanced and includes sufficient images for all classes. Additionally, using a lower learning rate for fine-tuning can help. If you haven't already, please update to the latest version of the Ultralytics package to benefit from recent improvements. If the problem persists, providing more details about your dataset and training process could help diagnose the issue further.

Hi @pderrenger, 
I’d like to ask: if training from a pre-trained model still requires the full dataset (old + new data, balanced), what is the advantage over training it from scratch?
I assume that if my model was trained on old classes A, B, and C with 500 images each, and then add classes D and E, I would still need 1500 + 500 + 500 images for fine-tuning. 
Thank you.

**Comment by pranta-barua007** [comment]
@pderrenger  is there any way to implement LwF(learning without forgetting) ? 
what i mean say i have 4 class dog, cat, cow, tiger. at first i will train on dog and cat class only , after that for cow and tiger i wont include data for dog and cat. and then the validation after all training done,  will held on validation set of those four classe.
Like this example -> https://github.com/pranta-barua007/learning-without-forgetting-pytorch
